adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDD.scala

```diff
@@ -85,6 +86,78 @@ sealed trait AlignmentRecordRDD extends AvroReadGroupGenomicRDD[AlignmentRecord,
   }
 
   /**
+   * Convert this set of reads into Coverage records
+   *
+   * @return Returns a CoverageRDD where all reads have been grouped and
+   * counted at all overlapping positions
+   */
+  def toCoverage: CoverageRDD = {
+    val coverage: RDD[(ReferenceRegion, Int)] =
+      rdd.rdd
+        .flatMap(r => {
+          val t: List[Long] = List.range(r.getStart, r.getEnd)
+          t.map(n => (ReferenceRegion(r.getContigName, n, n + 1), 1))
+        }).reduceByKey(_ + _)
+        .cache()
+        .sortByKey()
+        .mapPartitions(i => merge(i)) // merge adjacent regions with the same coverage
+
+    val features: RDD[Feature] = coverage.map(r => {
+      val fb = Feature.newBuilder()
+      fb.setContigName(r._1.referenceName)
+      fb.setStart(r._1.start)
+      fb.setEnd(r._1.end)
+      fb.setScore(r._2.toDouble)
+      fb.build()
+    })
+
+    CoverageRDD(FeatureRDD(features,
+      sequences))
+  }
+
+  /**
+   * Merges adjacent ReferenceRegions with the same value. This is used for merging coverage.
+   * @see toCoverage
+   *
+   * @param iter partition iterator of ReferenceRegion and coverage values
+   * @return merged tuples of adjacent ReferenceRegions and coverage
+   */
+  private def merge(iter: Iterator[(ReferenceRegion, Int)]): Iterator[(ReferenceRegion, Int)] = {
```

Comment:
Make `private[rdd]` and write unit tests. We'll really want to make sure this one is right...!